stdlib/Random/src/RNGs.jl

```diff
@@ -147,21 +147,32 @@ function show(io::IO, rng::MersenneTwister)
     end
     print(io, MersenneTwister, "(", repr(rng.seed), ", (")
     # state
-    adv = Integer[rng.adv_jump, rng.adv]
+    sep = ", "
+    show(io, rng.adv_jump)
+    print(io, sep)
+    show(io, rng.adv)
     if rng.adv_vals != -1 || rng.adv_ints != -1
+        print(io, sep)
         if rng.adv_vals == -1
             @assert rng.idxF == MT_CACHE_F
-            push!(adv, 0, 0) # "(0, 0)" is nicer on the eyes than (-1, 1002)
+            # "(0, 0)" is nicer on the eyes than (-1, 1002)
+            print(io, '0')
+            print(io, sep)
+            print(io, '0')
```

Comment:
Github won't let me highlight the whole `if`, but hopefully you get the idea:
```suggestion
            # "(0, 0)" is nicer on the eyes than (-1, 1002)
            (a, b) = (rng.adv_vals != -1) ? (rng.adv_vals, rng.idxF) : (Int64(0), 0)
            print(io, a)
            print(io, sep)
            print(io, b)
```